Code/Tools/AssetProcessor/native/AssetManager/PathDependencyManager.cpp

```diff
@@ -562,7 +566,11 @@ namespace AssetProcessor
                             //  2. When another product is created, all existing wildcard dependencies are compared against that product to
                             //  see if it matches them.
                             // This check here makes sure that the filter for 1 matches 2.
-                            if (!isExactDependency)
+                            if (!isExactDependency
+#if defined(CARBONATED)
+                                && !isRecursiveDependency
+#endif
+                                )
```

Comment:
Small thing, but this makes diffs difficult during engine updates difficult.
Do this instead:
```
#if defined(CARBONATED)
if (!isExactDependency && isRecursiveDependency)
#else
if (!isExactDependency)
#endif
```